### PR TITLE
Chunk files into batches of 1k

### PR DESCRIPTION
**Pull request by timgl**
COPY INTO command doesn't support more than 1k files. Haven't tested this in any way yet

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -624,20 +624,24 @@ async function copyIntoSnowflake({ cache, storage, global, jobs }: Meta<Snowflak
...
-    } catch {
-        await jobs
-            .retryCopyIntoSnowflake({ retriesPerformedSoFar: 0, filesStagedForCopy: filesStagedForCopy })
-            .runIn(3, 'seconds')
-        console.error(
-            `Failed to copy ${String(filesStagedForCopy)} from object storage into Snowflake. Retrying in 3s.`
-        )
+    const chunkSize = 999;
+    for (let i = 0; i < filesStagedForCopy.length; i += chunkSize) {
+        const chunkStagedForCopy = filesStagedForCopy.slice(i, i + chunkSize);
```
Comment (yakkomajuri) [inline index.ts]:
```suggestion
    // Snowflake's COPY INTO only supports up to 1000 files at a time, so we split into multiple COPY INTO statements of size 999
    const chunkSize = 999
    for (let i = 0; i < filesStagedForCopy.length; i += chunkSize) {
        const chunkStagedForCopy = filesStagedForCopy.slice(i, i + chunkSize)
```

**Review by yakkomajuri — APPROVED** [review]
would have loved to see a test, even if you just split that logic to another ~~file~~ function and write a simple unit test, but happy to get in for the sake of helping the customer if you feel this is very time sensitive

EDIT: file -> function

**Comment by timgl** [comment]
You’re right, I’ll write a test now!

File: index.test.ts [hunk 2]
```diff
@@ -112,6 +112,56 @@ test("handles events", async () => {
     expect(csvStrings.join()).toContain("456")
 })
 
+test("handles > 1k files", async () => {
+    // NOTE: we create random names for tests such that we can run tests
+    // concurrently without fear of conflicts
+    const bucketName = uuid4()
+    const snowflakeAccount = uuid4()
+
+    const snowflakeMock= jest.fn()
```
Comment (yakkomajuri) [inline index.test.ts]:
```suggestion
    const snowflakeMock = jest.fn()
```

File: index.test.ts [hunk 3]
```diff
@@ -112,6 +112,56 @@ test("handles events", async () => {
...
+        cache: cache,
+        storage: storage,
+        // Cast to any, as otherwise we don't match plugin call signatures
+        global: {snowflake: {copyIntoTableFromStage: snowflakeMock}} as any,
+        geoip: {} as any
+    }
+
+    await storage.set('_files_staged_for_copy_into_snowflake', Array(2100).fill('file'))
+    await cache.expire('lastRun', 0)
+   await snowflakePlugin.runEveryMinute(meta)
```
Comment (yakkomajuri) [inline index.test.ts]:
```suggestion
    await snowflakePlugin.runEveryMinute(meta) 
```

**Review by yakkomajuri — APPROVED** [review]
great, thanks!

**Pull request merged**